### PR TITLE
Rework init reaping code so it reaps only direct childs.

### PR DESCRIPTION
**Pull request by asac**
* introduce a background pid pool so we can reap them without stopping pantavisor
 * introduce tsh facilities to push and pop tracked pids
 * remember forked lxc processes that exit after daemonizing as background ids
 * remember tsh_run pids started in background in pid pool
 * also fix potential issues in run processes by resetting sigmask in child

### CODE REVIEW

File: init.c [hunk 1]
```diff
@@ -147,26 +147,11 @@ static int other_mounts()
 static pid_t waitpids(int *wstatus)
```
Comment (fortizc) [inline init.c:147]: This function is so small, maybe we can just remove it and call `waitpid(-1....` inside of the handler

File: utils/tsh.c [hunk 2]
```diff
@@ -65,6 +65,40 @@ static char **_tsh_split_cmd(char *cmd)
 	return ts;
 }
 
+pid_t bgpids[128];
```
Comment (fortizc) [inline utils/tsh.c:68]: The size (128) is used in both of the functions below, it should be a macro

File: init.c [hunk 3]
```diff
@@ -190,8 +175,10 @@ static void signal_handler(int signal)
 			       "Respawn of critical service failed %d: %s", pid,
 			       strerror(errno));
 		}
-		if (pv_pid == 0 || (pv_debug_is_ssh_pid(pid)))
+		if (getpid() == 1 || pv_debug_is_ssh_pid(pid) || tsh_bgid_pop(pid))
```
Comment (fortizc) [inline init.c:178]: The comment of this line is at the beginning of the loop (line 167) but it's outdated and incomplete

File: init.c (hunk 1 above)
Comment (anibalportero) [reply init.c]: done

File: utils/tsh.c (hunk 2 above)
Comment (anibalportero) [reply utils/tsh.c]: done

File: init.c (hunk 3 above)
Comment (anibalportero) [reply init.c]: done

**Pull request merged**